### ISSUE TITLE
Analyzer: change Enumerable.Concat with single element arrays into Append or Prepend

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/runtime#88314]
Since `Append` and `Prepend` comes much later than `Enumerable.Concat`, there are many old code using `Concat` with single-element array:
```csharp
values.Concat(new[] { moreValue })
```
which can be converted into:
```csharp
values.Append(moreValue)
```

`Prepend` is less common, and changes the order of two elements in code, but it would be more natural.

Category: Performance
Severity: Info

**Comment by campersau** [comment]
Note that `Enumerable.Append` is not directly supported in `net47` or older.
https://apisof.net/catalog/6ae58543-a1b1-3617-8780-227339489ae8

**Comment by huoyaoyuan** [comment]
> Note that `Enumerable.Append` is not directly supported in `net47` or older.

Yes, it needs extra attention for `netstandard2.0`, since it requires a shim on net461-net47.
For non-netstandard platforms, just detecting the existence of the methods should be sufficient.

**Comment by huoyaoyuan** [comment]
With collection expressions, there are actually additional options with spread: `[..values, morevalue]`. But maybe there are ones still prefer method call.
